### DISCUSSION TITLE
How is maestro studio able to know the exact id of all the UI elements just from the APK? Isn't this information supposed to be obfuscated by R8?

### DISCUSSION THREAD

**Discussion by sarimmehdi** [mobile-dev-inc/Maestro#839] [General]
Was very surprised to see this information available for all UI elements in maestro studio

**Comment by axelniklasson** [comment]
Hi @sarimmehdi! maestro doesn't read it from the APK, but instead we use UiAutomator and the Accessibility Framework to get this information.
